### ISSUE TITLE
Installer panic'd on partition table write

### ISSUE THREAD

**Issue by grahamwhaley** [clearlinux/clr-installer#476]
**Describe the bug**
Installing CLR v30390 live iso onto a NUC Skull i7-6770HQ with nvme SSD, the installer panic'd whilst trying to write the partition table:

![install_fail_small](https://user-images.githubusercontent.com/4761717/61468990-57748580-a976-11e9-87fb-f51581484600.JPG)

[clr-installer.log](https://github.com/clearlinux/clr-installer/files/3407161/clr-installer.log)

**To Reproduce**
1. Booted with EFI from USB stick with v30390 server live iso installed
1. Ran the clr-installer as root
1. Set up keyboard, country etc.
1. Did **not** delete the old partiton table afaik (sort of hard to tell actually in the UI)
1. Proceed with install, and it panic'd

**Expected behavior**
To Install as expected (iirc, the setup was 'wipe device', so I expected it to wipe the SSD completely)

**Screenshots**
Above.

**Environment (please complete the following information):**
 - Clear Linux OS Version: 30390. URLs default cdn.download.clearlinux.org/update
 - Installer Version 2.0.8
 - Platform: NUC skullcanyon i7-6770HQ. 
 I would have grabbed some nvme info, but I can't see any left in the dmesg, and I didn't find any tools on the installer that could read the part table off an nvme. I can update later if you need the info.

**Additional context**
Note, I have had difficulties installing this machine before on the nvme, but managed to convince Fedora28 or so to install last time.
Maybe this is similar/same to #182 
I'm tempted to wipe the part table clean and try an install, but will wait a short while before trying that in case you want me to grab more info or try something specific... (but, I won't wait forever, I'd quite like to move on to using the machine ;-) ).

**Note:** - github will not allow me to attach a yaml file directly. I'll paste here, and try to attach a renamed one as well...

[pre-install-clr-installer.yaml.txt](https://github.com/clearlinux/clr-installer/files/3407212/pre-install-clr-installer.yaml.txt)

```yaml
#clear-linux-config
#generated by clr-installer:2.0.8
#generated on: 2019-07-18_14:37:36_UTC
targetMedia:
- name: nvme0n1
  model: Samsung SSD 950 PRO 512GB
  majMin: "259:0"
  serial: S2GMNX0H718635L
  size: "512110190592"
  ro: "false"
  rm: "false"
  type: disk
  state: live
  children:
  - name: nvme0n1p1
    fstype: vfat
    mountpoint: /boot
    label: boot
    size: "150000000"
    ro: "false"
    rm: "false"
    type: part
  - name: nvme0n1p2
    fstype: swap
    label: swap
    size: "256000000"
    ro: "false"
    rm: "false"
    type: part
  - name: nvme0n1p3
    fstype: ext4
    mountpoint: /
    label: root
    size: "511704190592"
    ro: "false"
    rm: "false"
    type: part
keyboard: uk
language: en_US.UTF-8
bundles: [os-core, os-core-update, openssh-server]
telemetry: false
timezone: Europe/London
kernel: kernel-native
postArchive: true
hostname: skull
autoUpdate: true
copyNetwork: true
keepImage: true
```

Aha, the yaml has all the ssd and part info - so I may try a wipe/install sooner ;-)

**Comment by mdhorn** [comment]
The problem appears to be related to a warning after attempting to create the partition table:
`2019/07/18 14:37:32 [DBG] Warning: Partition(s) 2, 3 on /dev/nvme0n1 have been written, but we have been unable to inform the kernel of the change, probably because it/they are in use.  As a result, the old partition(s) will remain in use.  You should reboot now before making further changes.`

The partition table was not completely written out, then when we attempt to set the partitions type code with sgdisk it can't find or update that partition:
`4      An error prevented saving changes`

We need to look at how to ensure the new partition information is available before proceeding.

**Comment by mdhorn** [comment]
My guess would be this is probably related to the fact the partition table was msdos and we are rewriting to gpt and the kernel not seeing the change in partition table immediately.

**Comment by grahamwhaley** [comment]
In case it helps, the previous install on the disk was not under UEFI - I had to enable UEFI in the BIOS to try to install Clear.
If you want any more info from me @mdhorn then ask here, otherwise I'll try a wipe of the partition table and re-install next week.

**Comment by grahamwhaley** [comment]
fyi, a re-install with a 'full destructive' install worked, no problems. I'll leave it to the maintainers to decide if they can now close this Issue or not.
If you do need me to re-try something, do ask - and I will try, depending on machine availability.

**Comment by mdhorn** [comment]
@grahamwhaley Thank you for the update.
I'm going to leave this open as I think we have a problem here when the partition table is not GPT the kernel may not be getting the proper update before the code continues.

**Comment by mdhorn** [comment]
This should be resolved using the latest Advanced Disk configuration option enabled the latest release for the installer 2.2.0 in Clear Linux* OS release 31030.

Released in clr-installer version 2.2.0 in [Clear Linux* OS release 31030](https://cdn.download.clearlinux.org/releases/31030/clear/ clear-31030-live-desktop.iso.xz).
Latest release images are available here: https://clearlinux.org/downloads
